### DISCUSSION TITLE
Regarding JdbcStore managing two sequences in a single record.

### DISCUSSION THREAD

**Discussion by ydh6226** [quickfix-j/quickfixj#745] [Q&A]
Hi,

I'm curious about the JdbcStore logic to store the incoming_seqnum and outgoing_seqnum in the same record of the session table.

https://github.com/quickfix-j/quickfixj/blob/50d004661424eb3d1d07fe78a90cb91d85946c29/quickfixj-core/src/main/java/quickfix/JdbcStore.java#L117-L118

Since both the incoming and outgoing processes are trying to update the same record, this approach seems to induce a potential race condition and low throughput.

In contrast, FireStore keeps these sequences in separate files.

Is there a specific reason why JdbcStore does not follow a similar approach by keeping the two sequences in separate records?

Are there any advantages to the current logic that I am not aware of?

Or do you think there is no performance difference?

Thanks.

**Comment by chrjohn** [comment]
The reason for this is probably that no-one ever felt that this needs to be changed. 
There probably is a performance impact the way it is now.
Feel free to suggest changes in a PR. :)
